src/Core/Module/ModuleRepository.php

```diff
@@ -318,9 +318,25 @@ protected function addModulesFromHook(array $modules): array
      *
      * @return Module
      */
-    protected function enrichModuleAttributesFromHook(Module $module): ModuleInterface
+    protected function enrichModuleAttributesFromHook(Module $module, bool $getAvailableVersionsFromExternalSources = true): ModuleInterface
     {
-        $modulesFromHook = $this->getModulesFromHook();
+        // We don't have to get
+        if (!$getAvailableVersionsFromExternalSources) {
+            $coreModule = $module->getInstance();
+            if (null !== $coreModule && $coreModule->isRegisteredInHook('actionListModules')) {
+                $diskVersion = $this->getNextModuleVersion($module);
+                $modulesFromHook = [
+                    [
+                        'name' => $module->get('name'),
+                        'version' => $diskVersion,
+                        'version_available' => $diskVersion,
+                    ]
+                ];
+            }
+        } else {
+            $modulesFromHook = $this->getModulesFromHook();
+        }
+
         foreach ($modulesFromHook as $moduleFromHook) {
```

Comment:
Seems like`$modulesFromHook` can be used in a foreach without being defined, maybe you can give it a default `[]` value ?

Reply:
ah it's a draft, my bad !

Reply:
It's more a proposition than a draft
I'm ok with reviews 😉